cas-client-integration-jetty/src/main/java/org/jasig/cas/client/jetty/CasAuthenticator.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Licensed to Jasig under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Jasig licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.cas.client.jetty;
+
+import org.eclipse.jetty.security.Authenticator;
+import org.eclipse.jetty.security.ServerAuthException;
+import org.eclipse.jetty.server.Authentication;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.jasig.cas.client.Protocol;
+import org.jasig.cas.client.util.CommonUtils;
+import org.jasig.cas.client.util.ReflectUtils;
+import org.jasig.cas.client.validation.AbstractCasProtocolUrlBasedTicketValidator;
+import org.jasig.cas.client.validation.AbstractUrlBasedTicketValidator;
+import org.jasig.cas.client.validation.Assertion;
+import org.jasig.cas.client.validation.TicketValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Jetty authenticator component for container-managed CAS authentication.
+ * <p><em>NOTE:</em> This component does not support CAS gateway mode.</p>
+ *
+ * @author Marvin S. Addison
+ * @since 3.4.2
+ */
+public class CasAuthenticator extends AbstractLifeCycle implements Authenticator {
+
+    /** Name of authentication method provided by this authenticator. */
+    public static final String AUTH_METHOD = "CAS";
+
+    /** Session attribute used to cache CAS authentication data. */
+    private static final String CACHED_AUTHN_ATTRIBUTE = "org.jasig.cas.client.jetty.Authentication";
+
+    /** Logger instance. */
+    private final Logger logger = LoggerFactory.getLogger(CasAuthenticator.class);
+
+    /** Map of tickets to sessions. */
+    private final ConcurrentMap<String, WeakReference<HttpSession>> sessionMap =
+            new ConcurrentHashMap<String, WeakReference<HttpSession>>();
+
+    /** CAS ticket validator component. */
+    private TicketValidator ticketValidator;
+
+    /** Space-delimited ist of server names. */
```

Comment:
Nice catch. I had to read several times to see what you were pointing out, then finally saw the typo.